healthcheck/watch_test.go

```diff
@@ -23,10 +23,12 @@ func TestWatch(t *testing.T) {
 	require.Nil(t, watchers["testService"], "testService watcher")
 	Watch("testService", func(status Status) {})
 	require.NotNil(t, watchers["testService"], "testService watcher")
+	delete(watchers, "testService")
```

Comment:
`Watch` and `GetWatcher` depend on global variable `watchers`, which gives off a bad code smell.
Even with global variables, test cases should be independent of each other, at least apear to be so.
`delete`ing `testService` couple two test cases with the same key. I suggest using `t.Name()` as key to fix it.

Reply:
using `t.Name()` is a pretty good suggestion. But I think we still need `delete`ing operation at the end of test to make  it repeatable.